_posts/2020-09-01-leveraging-rust-to-build-cross-platform-mobile-libraries.md

```diff
@@ -0,0 +1,79 @@
+
+All "This Week in Glean" blog posts are listed in the [TWiG index](https://mozilla.github.io/glean/book/appendix/twig.html)
+(and on the [Mozilla Data blog](https://blog.mozilla.org/data/category/glean/)).
+This article is [cross-posted on the Mozilla Data blog](https://blog.mozilla.org/data/2020/08/24/this-week-in-glean-leveragin-rust/).
+
+---
+
+A couple of weeks ago I gave a talk titled ["Leveraging Rust to build cross-platform mobile libraries"][video].
+You can find my [slides as a PDF](https://fnordig.de/talks/2020/rustydays/slides.pdf).
+It was part of the [Rusty Days][rustydays] Webference, an online conference that was initially planned to happen in Poland, but had to move online.
+Definitely check out [the other talks][playlist].
+
+[rustydays]: https://rusty-days.org/
+[video]: https://www.youtube.com/watch?v=j5rczOF7pzg
+[slides]: https://fnordig.de/talks/2020/rustydays/slides.pdf
+[playlist]: https://www.youtube.com/watch?v=QaCvUKrxNLI&list=PLf3u8NhoEikhTC5radGrmmqdkOK-xMDoZ
+
+One thing I wanted to achieve with that talk is putting that knowledge out there.
+While multiple teams at Mozilla are already building cross-platform libraries, with a focus on mobile integration,
+the available material and documentation is lacking.
+I'd like to see better guides online, and I probably have to start with what we have done.
+But that should also be encouragement for those out there doing similar things to blog, tweet & speak about it.
+
+### Who else is using Rust to build cross-platform libraries, targetting mobile?
+
+I'd like to hear about it.
+Find me on Twitter ([@badboy\_](https://twitter.com/badboy_)) or drop me [an email](mailto:janerik@fnordig.de).
+
+## The Glean SDK
+
+I won't reiterate the full talk ([go watch it][video], really!), so this is just a brief overview of the Glean SDK itself.
+
+The Glean SDK is our approach to build a modern Telemetry library, used in Mozilla's mobile products and soon in Firefox on Desktop as well.
+
+The SDK consists of multiple components, spanning multiple programming languages for different implementations.
+All of the Glean SDK lives in the GitHub repository at [mozilla/glean](https://github.com/mozilla/glean).
+This is a rough diagram of the Glean SDK tech stack:
+
+![Glean SDK Stack](https://tmp.fnordig.de/blog/2020/glean-stack.png)
+
+On the very bottom we have `glean-core`, a pure Rust library that is the heart of the SDK.
+It's responsible for controlling the database, storing data and handling additional logic (e.g. assembling pings, clearing data, ..).
+As it is pure Rust we can rely on all Rust tooling for its development.
+We can write tests that `cargo test` picks up. We can generate the full API documentation thanks to `rustdoc`
+and we rely on `clippy` to tell us when our code is suboptimal.
+Working on `glean-core` should be possible by everyone that knows some Rust.
```

Comment:
```suggestion
Working on `glean-core` should be possible for everyone that knows some Rust.
```

Reply:
This makes me wish we had a "good-first-bug"/"good-next-bug" tag to link to here.